tsdb/index_test.go

```diff
@@ -159,7 +159,9 @@ func TestIndexSet_DedupeInmemIndexes(t *testing.T) {
 
 			var indexes []tsdb.Index
 			for i := 0; i < testCase.tsiN; i++ {
-				indexes = append(indexes, MustOpenNewIndex(tsi1.IndexName))
+				newIdx := MustOpenNewIndex(tsi1.IndexName)
+				indexes = append(indexes, newIdx)
+				defer newIdx.Close()
```

Comment:
Sneaking a fix in here to avoid leaving tmp files hanging around. There are a few other tests with this style of problem.